### PR TITLE
fix(lightspeed): numbered pointers text breaks into two chat  bubbles

### PR DESCRIPTION
**Pull request by ciiay**
## Hey, I just made a Pull Request!

For [RHDHBUGS-2988](https://redhat.atlassian.net/browse/RHDHBUGS-2988)

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/redhat-developer/rhdh-plugins/blob/main/CONTRIBUTING.md#creating-changesets))
- [ ] Added or Updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

image

### CODE REVIEW

**Review by HusneShabbir — APPROVED** [review]
/lgtm
/approve
works as expected and have tested all the edge cases

https://github.com/user-attachments/assets/9b79ab1d-4a13-454b-8c85-7d4b3aca2b2f

https://github.com/user-attachments/assets/5dea9139-1ab7-4256-b343-ee4dcb02633b

File: workspaces/lightspeed/plugins/lightspeed/src/utils/lightspeed-chatbox-utils.tsx [hunk 1]
```diff
@@ -42,6 +42,104 @@ export const getTimestampVariablesString = (v: number) => {
...
+ * Trims user chat input and reduces blank lines that split Markdown blocks
+ * (e.g. paragraph + blank line + ordered list) so PatternFly renders one user bubble.
+ */
+export const normalizeChatUserInput = (input: string): string => {
+  let s = input.replace(/\r\n/g, '\n').trim();
+  if (!s) {
+    return s;
+  }
+  s = s.replace(/\n{3,}/g, '\n\n');
+  s = s.replace(/\n\n(?=\d+\.\s)/gm, '\n');
```
Comment (karthikjeeyar) [inline workspaces/lightspeed/plugins/lightspeed/src/utils/lightspeed-chatbox-utils.tsx]:
```suggestion
  s = s.replace(/\n\n(?=\s*\d+\.\s)/gm, '\n');
```

When you have a leading space in-front of ordered list, this still breaks.
```
Explain me about

  1. RHDH
  2. Tekton
```

Image

Comment (karthikjeeyar) [reply workspaces/lightspeed/plugins/lightspeed/src/utils/lightspeed-chatbox-utils.tsx]: Also can you add a test to cover this?

File: workspaces/lightspeed/plugins/lightspeed/src/utils/lightspeed-chatbox-utils.tsx [hunk 2]
```diff
@@ -42,6 +42,104 @@ export const getTimestampVariablesString = (v: number) => {
...
+
+const escapeBulletListMarker = (line: string) =>
+  line.replace(/^(\s*)([*+-])(\s)/, '$1\\$2$3');
+
+/**
+ * Paragraph + ordered list (even with a single newline) parses as two blocks; merge
+ * intro + contiguous numbered lines into one paragraph using hard breaks (two spaces
+ * + newline) and escaped `1.` markers so Markdown stays a single block.
+ */
+const foldOrderedListWithIntro = (s: string): string => {
```
Comment (karthikjeeyar) [inline workspaces/lightspeed/plugins/lightspeed/src/utils/lightspeed-chatbox-utils.tsx]: nit:

There are some code duplication between `foldOrderedListWithIntro` and `foldBulletListWithIntro`. Consider moving having a generic function to reduce some duplication.

File: workspaces/lightspeed/plugins/lightspeed/src/utils/lightspeed-chatbox-utils.tsx (hunk 1 above)
Comment (karthikjeeyar) [reply workspaces/lightspeed/plugins/lightspeed/src/utils/lightspeed-chatbox-utils.tsx]: Similar issue is observed in bullet lists as well.

**Review by karthikjeeyar — CHANGES_REQUESTED** [review]
Added Some observations/ideas


I also noticed a minor issue which is not related to this PR change, but when adding empty spaces in the message bar enables the send button (Ideally it should not enable it), when I click on send it is not sending the message which is good. But user doesn't have any visual clue after the press the send button. Its better to disable the send button when user just type whitespaces.

**Review by HusneShabbir — COMMENTED** [review]
https://github.com/user-attachments/assets/74026f94-093c-4ff4-a917-e8ddb6df98c5

Thanks for fixing.

**Review by karthikjeeyar — APPROVED** [review]
/approve
/lgtm
